iceoryx_hoofs/include/iceoryx_hoofs/cxx/generic_raii.hpp

```diff
@@ -42,39 +41,46 @@ namespace cxx
 ///     // destructor
 /// }
 /// @endcode
-class GenericRAII
+template <uint64_t CleanupCapacity = DEFAULT_FUNCTION_CAPACITY>
+class GenericRAIIWithVariableCapacity
```

Comment:
The class documentation needs to be updated since it still mentions `GenericRAII`.
For a long time I thought about renaming this to `ScopeGuard` and since there is an alias further down, what do you think about using this name for the template?

Reply:
Sure, but `ScopeGuard` is already kind of established terminology for this kind of type, e.g. Andrei Alexandrescu also uses this term https://www.drdobbs.com/cpp/generic-change-the-way-you-write-excepti/184403758?pgno=2

But it's okay if we keep it like it is in this PR but to use common terminology I still would like to rename it in the future ... let's see if this ever happens :sweat_smile: